jni/src/main/java/com/horizen/merkletreenative/BigMerkleTree.java

```diff
@@ -26,42 +28,50 @@ public static BigMerkleTree init(int height, String statePath, String dbPath, St
     // Returns the position to which insert the leaf
     public long getPosition(FieldElement leaf) {
         if (merkleTreePointer == 0)
-            throw new IllegalArgumentException("merkleTreePointer must be not null.");
+            throw new IllegalArgumentException("BigMerkleTree instance was freed.");
         return nativeGetPosition(leaf);
     }
 
     private native boolean nativeIsPositionEmpty(long position);
 
     public boolean isPositionEmpty(long position){
         if (merkleTreePointer == 0)
-            throw new IllegalArgumentException("merkleTreePointer must be not null.");
+            throw new IllegalArgumentException("BigMerkleTree instance was freed.");
         return nativeIsPositionEmpty(position);
     }
 
     private native void nativeAddLeaf(FieldElement leaf, long position);
 
     public void addLeaf(FieldElement leaf, long position){
```

Comment:
So, what about this method? Are you going to remove `position` arg?

Reply:
I'm not sure about this one. From an usage perspective can be misleading, but probably for testing and debugging it will be useful